src/main/java/cardibuddy/model/deck/DeckContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,31 @@
+package cardibuddy.model.deck;
+
+import cardibuddy.commons.util.StringUtil;
+import cardibuddy.model.flashcard.Flashcard;
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
+public class DeckContainsKeywordsPredicate implements Predicate<Deck> {
+    private final List<String> keywords;
+
+    public DeckContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Deck deck) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(deck.getTitle().toString(), keyword));
```

Comment:
Actually, is it possible to make the title be a 'String' instead of a 'Title' in the Deck Class? Then we won't need to toString() here.

Reply:
Hmmm but in the Person class Name and the other attributes were separate classes